notebooks/xgboost-model/xgboostDmatrixExample.ipynb

```diff
@@ -0,0 +1,558 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "certifai_toolkit_path = 'path_to_certifai_toolkit'\n",
+    "!find $certifai_toolkit_path/packages/all       -type f ! -name \"*console-*\" ! -name \"*client-*\" | xargs -I % sh -c 'pip install % ' ;\n",
+    "!find $certifai_toolkit_path/packages/python3.6 -type f   -name \"*engine-*\"                      | xargs -I % sh -c 'pip install % ' ;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'1.2.13'"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# check version of certifai installed\n",
+    "from certifai.scanner.version import  get_version\n",
+    "get_version()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# imports for building certifai scan\n",
+    "from certifai.scanner.builder import (CertifaiScanBuilder, CertifaiPredictorWrapper, CertifaiModel, CertifaiModelMetric,\n",
+    "                                      CertifaiDataset, CertifaiGroupingFeature, CertifaiDatasetSource,\n",
+    "                                      CertifaiPredictionTask, CertifaiTaskOutcomes, CertifaiOutcomeValue)\n",
+    "from certifai.scanner.report_utils import scores, construct_scores_dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### create a CertifaiPredictiorWrapper object from transformed model created above\n",
+    "\n",
+    "- this predictiorWrapper object will be used by certifai to perform model predicts as constructor to CertifaiModel\n",
+    "- run the assert test below to confirm predictions from raw model and certifaiWrapped model are identical"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xbg_certifai_wrapped_model = CertifaiPredictorWrapper(transformed_model,encoder=encoder,decoder=decoder)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# test to assert wrapped certifai model predicts == raw model predicts\n",
+    "assert (xbg_certifai_wrapped_model.model.predict(X_test.values) == \n",
+    "        decoder(xg_reg.predict(xgb.DMatrix(encoder(X_test.values))))).all"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### creating a certifai evaluation dataset\n",
+    "\n",
+    "- earlier we modified our dataset to frop certain non useful columns\n",
+    "- and we ran our encoder on the cleaned dataset\n",
+    "- we will pass the cleaned dataframe (with the removed columns) to certifai for evaluation\n",
+    "- this is needed since the dropped columns are non encoded and are completely useless for certifai evaluation"
```

Comment:
It's needed because the model doesn't expect those columns, right?

Reply:
correct but even before that, these columns are not encoded. model also doesn't use the `outcome` or target column but we need that in the dataset for metrics calculation purposes